packages/expo/src/modules/register/registerFirstStep/Register.tsx

```diff
@@ -0,0 +1,236 @@
+
+const validationSchema = Yup.object().shape({
+  name: Yup.string().required('O campo nome é obrigatório!'),
+  race: Yup.string().required('O campo raça é obrigatório!'),
+  observations: Yup.string().required('O campo observações é obrigatório!'),
+  gender: Yup.string().required('É obrigatório escolher o sexo do animal!'),
+  animalsSize: Yup.string().required('É obrigatório escolher um tipo de animal!'),
+  animalsType: Yup.string().required('Escolha o tamanho do animal!'),
+});
+
+const LARGE = 'large';
+
+interface Values {
+  gender: string;
+  observations: string;
+  race: string;
+  name: string;
+  age: number;
+  animalsType: string;
+  animalsSize: string;
+}
+
+interface Props {
+  onSubmit: (values: Values) => void;
+}
+
+const Register: React.FC<Props> = ({onSubmit}) => {
+  const [age, setAge] = useState(1);
+  const navigation = useNavigation();
+  const [animalsType, setAnimalsType] = useState([
+    {value: 'Cao', checked: false},
+    {value: 'Gato', checked: false},
+    {value: 'OUTRO', checked: false},
+  ]);
+  const [animalsSize, setAnimalsSize] = useState([
+    {value: 'PEQUENO', checked: false},
+    {value: 'MÉDIO', checked: false},
+    {value: 'GRANDE', checked: false},
+  ]);
+
+  function handleSelectedAnimalType(animal: string) {
+    const animalsTypeUpdated = [...animalsType];
+
+    animalsTypeUpdated.forEach(animalUpdated => {
+      if (animalUpdated.value === animal) {
+        animalUpdated.checked = true;
+      } else {
+        animalUpdated.checked = false;
+      }
+    });
+
+    setAnimalsType(animalsTypeUpdated);
+  }
+
+  function handleSelectedAnimalSize(animal: string) {
+    const animalsSizeUpdated = [...animalsSize];
+
+    animalsSizeUpdated.forEach(animalUpdated => {
+      if (animalUpdated.value === animal) {
+        animalUpdated.checked = true;
+      } else {
+        animalUpdated.checked = false;
+      }
+    });
+
+    setAnimalsSize(animalsSizeUpdated);
+  }
+
+  function nextScreen(errors) {
+    {
+      errors ? '' : navigation.navigate('RegisterSecondStep');
+    }
+  }
+
+  return (
+    <Background>
+      <Wrapper>
+        <Header />
+        <Formik<Values>
+          initialValues={{name: '', race: '', observations: '', gender: '', age, animalsType: '', animalsSize: ''}}
+          validationSchema={validationSchema}
+          onSubmit={value => {
+            onSubmit(value);
+          }}>
+          {({handleChange, handleSubmit, values, errors, setFieldValue}) => (
+            <>
+              <ScrollViewContainer>
+                {registerForm.map((field, index) => (
+                  <>
+                    <Label>{field.label}</Label>
+                    <Input
+                      key={index}
+                      placeholder={field.placeholder}
+                      onChangeText={handleChange(field.name)}
+                      value={values[field.name]}
+                    />
+                    {errors && <Error>{errors[field.name]}</Error>}
```

Comment:
I think that makes sense you render the `Error` component inside `Input` and pass a condition in order to render it conditionally.